modules/admin/ref-rbac-rest-api-endpoints.adoc

```diff
@@ -1,7 +1,209 @@
 [id='ref-rbac-rest-api-endpoints_{context}']
 = Supported RBAC REST API endpoints
 
-The RBAC REST API provides the following endpoints for managing permission policies and roles in the {product-short} and for retrieving information about the policies and roles.
+The RBAC REST API provides the following endpoints for managing roles, permissions, and conditional policies in the {product-short} and for retrieving information about the roles and policies.
+
+== Roles
+
+The RBAC REST API supports the following endpoints for managing roles in the {product}.
+
+[GET] /api/permission/roles::
++
+--
+Returns all roles in {product-short}.
+
+.Example response (JSON)
+[source,json]
+----
+[
+  {
+    "memberReferences": ["user:default/username"],
+    "name": "role:default/guests"
+  },
+  {
+    "memberReferences": [
+      "group:default/groupname",
+      "user:default/username"
+    ],
+    "name": "role:default/rbac_admin"
+  }
+]
+----
+--
+
+[GET] /api/permission/roles/{kind}/{namespace}/{name}::
++
+--
+Returns information for a single role in {product-short}.
+
+.Example response (JSON)
+[source,json]
+----
+[
+  {
+    "memberReferences": [
+      "group:default/groupname",
+      "user:default/username"
+    ],
+    "name": "role:default/rbac_admin"
+  }
+]
+----
+--
+
+[POST] /api/permission/roles/{kind}/{namespace}/{name}::
++
+--
+Creates a role in {product-short}.
+
+.Request parameters
+[cols="15%,45%,15%,25%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`body`
+|The `memberReferences`, `group`, `namespace`, and `name` the new role to be created.
+|Request body
+|Required
+|===
+
+.Example request body (JSON)
+[source,json]
+----
+{
+  "memberReferences": ["group:default/test"],
+  "name": "role:default/test_admin"
+}
+----
+
+.Example response
+[source]
+----
+201 Created
+----
+--
+
+[PUT] /api/permission/roles/{kind}/{namespace}/{name}::
++
+--
+Updates `memberReferences`, `kind`, `namespace`, or `name` for a role in {product-short}.
+
+.Request parameters
+The request body contains the `oldRole` and `newRole` objects:
+[cols="15%,45%,15%,25%", frame="all", options="header"]
+|===
+|Name
+|Description
+|Type
+|Requirement
+
+|`body`
+|The `memberReferences`, `group`, `namespace`, and `name` the new role to be created.
+|Request body
+|Required
+|===
+
```

Comment:
We have a table name (_Table 2. Request parameters_) above the previous table but not above this table. Do we need to add a table heading to this one, as well?

Reply:
I don't see any table without a caption. Could you please name the endpoint maybe?